src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = 0;
+        int buy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    continue;
+                }
```

Lines 17-20: The code currently does
```
String[] parts = line.split(",");
if (parts.length != 2) {
    continue;
}
```
This violates the checklist item: "Validate that each line splits into exactly two parts; otherwise throw a RuntimeException as per exception handling requirement." Instead of silently continuing, throw a RuntimeException with an informative message (include the filename and the invalid line). See checklist guidance for validation and not ignoring exceptions .